### ISSUE TITLE
Org Monitoring Workflow failing at sf hardis:org:monitor:backup with No default environment found

### ISSUE THREAD

**Issue by MattFaz** [hardisgroupcom/sfdx-hardis#822]
Hi,

I've setup the connected app and configured the Github secrets accordingly.
However at the `Login & Retrieve Metadata` step I'm receiving the following error: 

```sh
Monitoring sfdx-hardis: Metadata Backup for "monitoring_squareinc"
[sfdx-hardis][command] git rev-parse --show-toplevel
[sfdx-hardis] hardis:auth:login execution time 0:00:00.355
[sfdx-hardis][command] git rev-parse --show-toplevel
[sfdx-hardis][command] sf org display --target-org monitoring_squareinc --json
Error (NoDefaultEnvError): No default environment found. Use -o or --target-org to specify an environment.
Error: Process completed with exit code 1.
```

![image](https://github.com/user-attachments/assets/26db8f1c-b56e-4fa8-8d06-1fb8accadd4b)

I added a `echo login success` after the `sf hardis:auth:login` command and I appear to be logging successfully.
The issue lies in the `sf hardis:org:monitor:backup` command, I've tried adding a target org as detailed in the [docs](https://sfdx-hardis.cloudity.com/hardis/org/monitor/backup/) with `sf hardis:org:monitor:backup -o $CONFIG_BRANCH` but same error.

Any help would be appreciated. Below is my `.yml` file _(removed all steps after failing step to reduce size)_.
```yml
on:
  push:
  schedule:
    - cron: "0 0 * * *"
  workflow_dispatch:

name: Org Monitoring sfdx-hardis

permissions: read-all

jobs:
  ##############################################
  ### Sfdx Sources Backup + Push new commit ####
  ##############################################
  backup:
    runs-on: ubuntu-latest
    name: Backup metadatas
    permissions: write-all
    strategy:
      fail-fast: false
      max-parallel: 10
      matrix:
        # MANUAL: Add your monitored git branches here
        branch:
          - monitoring_squareinc
    steps:
      # Checkout repo
      - name: Checkout code
        uses: actions/checkout@v4
        with:
          fetch-depth: 0 # Faster code checkout fetching only latest commit
          ref: ${{ matrix.branch }}
          persist-credentials: true
      # Setup node
      - name: Setup Node
        uses: actions/setup-node@v4
        with:
          node-version: "20"
      # SFDX & plugins
      - name: Install SFDX and plugins
        run: |
          npm install --no-cache @salesforce/cli --global
          sf plugins install --force @salesforce/plugin-packaging
          echo 'y' | sf plugins install --force sfdx-hardis
          echo 'y' | sf plugins install --force sfdx-git-delta
          sf version --verbose --json
      # Login & check deploy with test classes & code coverage
      - name: Login & Retrieve Metadata
        env:
          # MANUAL: Update variables below to add authentication variables you need
          SFDX_CLIENT_ID_MONITORING_SQUAREINC: ${{ secrets.SFDX_CLIENT_ID_MONITORING_SQUAREINC }}
          SFDX_CLIENT_KEY_MONITORING_SQUAREINC: ${{ secrets.SFDX_CLIENT_KEY_MONITORING_SQUAREINC }}
          SFDX_DEPLOY_WAIT_MINUTES: 120 # Override if necessary
          SFDX_TEST_WAIT_MINUTES: 120 # Override if necessary
          CI_COMMIT_REF_NAME: ${{ matrix.branch }}
          ORG_ALIAS: ${{ matrix.branch }}
          CONFIG_BRANCH: ${{ matrix.branch }}
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          echo "Monitoring sfdx-hardis: Metadata Backup for \"$CONFIG_BRANCH\""
          sf hardis:auth:login
          sf hardis:org:monitor:backup
```

**Comment by nvuillam** [comment]
@MattFaz usually, when the login fails, configuring it again is faster than looking for the cause :) (which can be a typo, a space added in a variable...)

Run again "Configure org monitoring" , commit, push and i'm pretty sure it will work :)

Note: i'll work anyway on a better output than the current "no output" when login has not been performed

**Comment by MattFaz** [comment]
Thanks for the response @nvuillam .

When running the _'Configure Org Monitoring'_ option the deployment of the connected app fails due to the hardcoded wait time of 60 minutes _(our tests take ~90mins)_. Throughout my process of debugging I've used the 'Configure Org Monitoring' a few times and it turns out I missed the message stating I need to update the `.crt` file on the managed app.

Once I updated that it is now working :)

One recommendation from me would be to make the wait time on the connected app deployment be 120 _(or configurable)_ like it is for other commands.

**Comment by nvuillam** [comment]
@MattFaz thanks for your feedback, and glad it works now on your org :)
 i reopen the issue until i increase the waiting time and add a warning message :)

**Comment by nvuillam** [comment]
@MattFaz released in 5.1.0 :)
